### DISCUSSION TITLE
programData.json addition broke OpenSprinkler-[mobile]-app:-(

### DISCUSSION THREAD

**Discussion by mw46d** [Dan-in-CA/SIP#254] [Q&A] [answered]
It looks like the changes related to that new json file/structure are not understood by the OpenSprinkler app:-( I did not look too much into the app side, but it looks like a restored support would require a two-way conversion between the new & old program[s] structures:-(

A different/related issue, but easier fixed: the auto-conversion does not work for installations with extension boards.

**Comment by Dan-in-CA** [comment, marked as answer]
Yes, the new file format is no longer compatible with the OpenSprinkler mobile app. That caused a long delay in implementing the change. 
Of course the change in program data format involved extensive changes to SIP's internal code, both the core Python code and the JavaScript that implements the web UI. The new format is much easier to understand and work with.
The mobile app never did provide access to SIP's plugin system which is essential for many users. It would not be practical to try and develop a two-way translation system for the mobile app.

SIP's web UI has been updated to make it mobile friendly. Try accessing SIP from the browser on your phone or other mobile device. From the browser you can make a desktop shortcut for SIP that will allow one tap access. Without requiring the installation of an app, SIP does not take up a lot of storage space on a mobile device just the shortcut and some browser cache.

I will look into the problem with conversion of extension board data. In the mean time please check **Options > Station handling > Station extensions** to see if the correct number of extensions is set. For 1 extension 8 stations are added. The snames.json file that holds station names and sd.json file that holds general settings were not changed and the original programs.json file was left unchanged so all data is preserved.

This subject is of general interest so I will move it to Discussions. Please check there for updates.

**Reply by Dan-in-CA** [reply]
I found the problem with the program data converter (convert_progs.py).
An update has been posted to GitHub in SIP version 4.1.11 (2021-03-30).

Dan
